### PR TITLE
Update docs on docs.

### PR DESCRIPTION
**Pull request by anntzer**
Update PULL_REQUEST_TEMPLATE.md to emphasize markup specificities in RST
(as opposed to md) regarding double/single backquotes and asterisks.
(I'm a bit tired of pointing to this every other docstring PR.)

In documenting_mpl.rst:
- Various reformattings/line wrappings/rewordings.
- Explicitly mark the highlighting language of each code block, as we
  have a mix of python/shell/rst (and emacs lisp).
- Drop unneeded `:func:`/`:class:` markups.
- Drop request to line wrap docstrings at 70 characters (in practice
  they are line-wrapped at 79 like normal Python sources).
- Do not forbid spanned cells in tables, which recent Sphinxes handles
  just fine.

In make.py:
- Simplify argument parsing, slightly improving output of `./make.py
--help`

## PR Checklist

- [ ] Has Pytest style unit tests
- [ ] Code is PEP 8 compliant 
- [ ] New features are documented, with examples if plot related
- [ ] Documentation is sphinx and numpydoc compliant
- [ ] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [ ] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

**Comment by dstansby** [comment]
Would be good if someone could double check `make.py`..

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -6,6 +6,12 @@ https://matplotlib.org/devdocs/devel/index.html-->
 example "Raises ValueError on Non-Numeric Input to set_xlim".  Please avoid
 non-descriptive titles such as "Addresses issue #8576".-->
 
+<!--If you are contributing fixes to docstrings
+(which are very welcome!), please pay attention to
```
Comment (tacaswell) [inline .github/PULL_REQUEST_TEMPLATE.md]: Can we make this less passive aggressive?

Comment (anntzer) [reply .github/PULL_REQUEST_TEMPLATE.md]: Feel free to propose an alternate wording 😃

**Comment by anntzer** [comment]
Temporarily closed to let #9513 be merged first.

File: doc/devel/documenting_mpl.rst [hunk 2]
```diff
@@ -10,75 +10,73 @@ Getting started
...
+1. Sphinx 1.3 or later (1.5.0 is not supported)
 2. numpydoc 0.4 or later
 3. IPython
-4. Mock
+4. mock
 5. colorspacious
-6. pillow
-7. graphviz
+6. Pillow
+7. Graphviz
```
Comment (dstansby) [inline doc/devel/documenting_mpl.rst]: Needs `sphinx-gallery`

File: doc/devel/documenting_mpl.rst [hunk 3]
```diff
@@ -10,75 +10,73 @@ Getting started
...
+directories that contain more ``.rst`` files, and configuration files for
+Sphinx_.
 
 .. note::
 
-  An exception to this are the directories :file:`gallery` and :file:`tutorials`, which
-  exist in the root directory. These contain Python files that are built by `Sphinx Gallery`_.
-  When the docs are built, directories of the same name will be generated inside of :file:`docs/`.
-  The generated directories :file:`docs/gallery` and :file:`docs/tutorials` can be safely deleted.
+   An exception to this are the directories :file:`gallery` and
```
Comment (dstansby) [inline doc/devel/documenting_mpl.rst]: Should be `examples` instead of `gallery` I think now?

File: doc/devel/documenting_mpl.rst [hunk 4]
```diff
@@ -10,75 +10,73 @@ Getting started
...
+   :file:`tutorials`, which exist in the root directory.  These contain Python
+   files that are built by `Sphinx Gallery`_.  When the docs are built,
+   directories of the same name will be generated inside of :file:`docs/`.  The
+   generated directories :file:`docs/gallery` and :file:`docs/tutorials` can be
```
Comment (dstansby) [inline doc/devel/documenting_mpl.rst]: `examples` instead of `gallery` again here

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 1 above)
Comment (dstansby) [reply .github/PULL_REQUEST_TEMPLATE.md]: I'm worried that the template is getting too large for anyone to bother reading - I'm happy to just get rid of (which are very welcome!) here,

File: doc/devel/documenting_mpl.rst (hunk 2 above)
Comment (anntzer) [reply doc/devel/documenting_mpl.rst]: Addressed in #9513.

File: doc/devel/documenting_mpl.rst (hunk 3 above)
Comment (anntzer) [reply doc/devel/documenting_mpl.rst]: No, the files end up in the gallery subfolder of the source tree.

**Comment by anntzer** [comment]
Removed the 'very welcome'.

**Comment by jklymak** [comment]
This looks fine to me.  I was disappointed to *still* not see a description of how to reference functions both in MPL and using inter-sphinx.  This is badly needed in my opinion.  @dstansby does your PR address this?

**Comment by dstansby** [comment]
It doesn't yet, but it can!

**Comment by tacaswell** [comment]
re-milestoning this to 2.2.  There is at least one other PR that did not get backported previously, the backport is not clean at all.

**Pull request merged**